packages/phishing-controller/src/PhishingController.ts

```diff
@@ -861,6 +909,154 @@ export class PhishingController extends BaseController<
     return combinedResponse;
   };
 
+  /**
+   * Map chain ID to chain name for the API.
+   *
+   * @param chainId - The chain ID.
+   * @returns The chain name.
+   */
+  readonly #chainIdToName: Record<string, string> = {
+    '0x1': 'ethereum',
+    '0x89': 'polygon',
+    '0x38': 'bsc',
+    '0xa4b1': 'arbitrum',
+    '0xa86a': 'avalanche',
+    '0x2105': 'base',
+    '0xa': 'optimism',
+    '0x76adf1': 'zora',
+    '0xe708': 'linea',
+    '0x27bc86aa': 'degen',
+    '0x144': 'zksync',
+    '0x82750': 'scroll',
+    '0x13e31': 'blast',
+    '0x74c': 'soneium',
+    '0x79a': 'soneium-minato',
+    '0x14a34': 'base-sepolia',
+    '0xab5': 'abstract',
+    '0x849ea': 'zero-network',
+    '0x138de': 'berachain',
+    '0x82': 'unichain',
+    '0x7e4': 'ronin',
+    '0x127': 'hedera',
+  };
+
+  /**
+   * Scan multiple tokens for malicious activity in bulk.
+   *
+   * @param request - The bulk scan request containing chainId and tokens.
+   * @param request.chainId - The chain ID in hex format (e.g., '0x1' for Ethereum).
+   * @param request.tokens - Array of token addresses to scan.
+   * @returns A mapping of lowercase token addresses to their scan results. Tokens that fail to scan are omitted.
+   */
+  bulkScanTokens = async (
```

Comment:
This method should be broken down into smaller methods. It currently seems to be doing too much: identify which tokens are on cache and which ones we need to fetch, handles all the fetch logic, populates the cache from the fetch ones. It can still be the parent method, but if we move some of its logic into its own methods, we will reduce cyclomatic complexity and improve its readability